doc/md/extension-api.md

```diff
@@ -0,0 +1,186 @@
+---
+id: extension-api
+title: Extension API
+---
+
+### Introduction
+
+The Ent [Extension API](https://github.com/ent/ent/blob/a19a89a141cf1a5e1b38c93d7898f218a1f86c94/entc/entc.go#L197)
+facilitates that creation of code-generation extensions that bundle together [Code-gen Hooks](code-gen.md#code-generation-hooks),
+[Templates](templates.md) and [Annotations](templates.md#annotations) to create reusable components
+that add new rich functionality to Ent's core.
```

Comment:
Maybe reference the entgql extension as an example?  https://pkg.go.dev/entgo.io/contrib/entgql#Extension